ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh

```diff
@@ -229,30 +264,117 @@ then
 fi
 
 if [ -e /root/bm.json ] ; then
-    cat /root/dev-scripts-additional-config >> /root/dev-scripts/config_root.sh
-
-    # On baremetal clusters DNS has been setup so that the clustername is part of the provision host long name
+    # On lab baremetal clusters DNS has been setup so that the clustername is part of the provision host long name
     # i.e. hostname == host1.clusterXX.ocpci.eng.rdu2.redhat.com
-    export LOCAL_REGISTRY_DNS_NAME=\$(hostname -f)
-    export "CLUSTER_NAME=\$(hostname -f | cut -d . -f 2)"
-
+    if [[ \$(hostname -f) =~ ocpci.eng.rdu2.redhat.com ]] ; then
```

Comment:
I'd suggest to introduce a switch and two different functions, `setup_internal_lab` and `setup_moc`, where to centralize the code for each environment. Just to slightly help in maintenance and readability, otherwise this part will become pretty hard to maintain. In addition, I guess that most of the hard-codings will be moved into some cluster profile config file

Reply:
My intent is to provide the values for most of the values hardcoded  in the cir extra data and then no longer have them hardcoded, I can't do this until the change to the format of "extra" is changes 
see  `echo "{\"nodes\":$(cat $NODESFILE)}" > $EXTRAFILE` above

once if done this I will be able to remove a lot of the env specific if statements, 

would this be ok, or would you prefer I add the env specific functions now and remove them later?